### DISCUSSION TITLE
How to pass JWT for RLS along standard postgres query

### DISCUSSION THREAD

**Discussion by bradcypert** [supabase/supabase#11512] [Questions]
Hi, sorry if this is a silly question but Im having a hard time figuring it out: How would I go about passing a JWT along for RLS policies from my API server (with a direct postgres connection).

My architecture is as follows:
I have a Flutter app that uses Supabase for auth. Once the user authenticates, all requests go through my API layer in Rust. I pass along the user's JWT to my API (that part's easy) but once I have the JWT token, I am not sure how to pass that along in a standard postgres query. On the rust side, I am using Diesel as a data abstraction layer, but can use Raw SQL queries if necessary.

Does anyone know what my postgres queries need to include to pass along the JWT token with each query to support RLS?

I do specifically want to go through my API and not communicate directly to Supabase from Flutter.

**Comment by GaryAustin1** [comment]
You used to be able to do something like this in your SQL:
```
set session role authenticated;
set request.jwt.claims to '{"role":"authenticated", "sub":"5950b438-b07c-4012-8190-6ce79e4bd8e5"}';
```
But as shown in the link, with the new postgres user role without super user privileges the first line does not work anymore.
https://github.com/supabase/supabase/discussions/9314#discussioncomment-3983254

You would need add logic to decode your jwt using the jwt secret to get the claims to put in the 2nd line.

Note this all occurs without the PostgREST processing of the role and the jwt.  If your RLS uses either authenticated or anon roles in the TO field of the policy you need the first line also.

**Reply by steve-chavez** [reply]
To add on to this. You need to make sure the settings are local and transaction scoped, so:

```
begin;
set local role authenticated;
set local request.jwt.claims to '{"sub":"5950b438-b07c-4012-8190-6ce79e4bd8e5"}';
<the_query>
end;
```

The claims come from the decoded JWT.

**Reply by GaryAustin1** [reply]
@steve-chavez 
The problem is with the new security changes these types of things don't work anymore.   I've added issues to the long security thread, but it is not possible to test RLS either this way or with the new explain in PostgREST because we cant set the flag to turn it on.

![image](https://user-images.githubusercontent.com/54564956/211353640-ae724fa4-2cc7-496d-bf11-6c642fbf5462.png)
